.github/workflows/macos.yaml

```
@@ -3,36 +3,42 @@ name: macos
 on: [push, pull_request]
```

Since we are already updating this file, it is best to update the push trigger to only work on the main branch. This will help decrease the GitHub action runs on each PR.

```suggestion
on:
  push:
    branches:
    - 'main'
  pull_request:
    branches:
    - '*'
```